### PR TITLE
LPS-168338 [Questions] Adjust the list of answers

### PR DESCRIPTION
**Pull request by juliocezar96**
[LPS-168338](https://issues.liferay.com/browse/LPS-168338)

### CODE REVIEW

**Comment by brenobcos** [comment]
Just starting reviewing :)

File: modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/ActivityQuestionRow.es.js [hunk 1]
```diff
@@ -57,6 +58,19 @@ export default function ActivityQuestionRow({
...
+				isBestAnswer: true,
+				label: Liferay.Language.get('best answer'),
+				symbol: 'check-circle-full',
+				text: stripHTML(question.articleBody).replace(
+					MESSAGE_TYPES.answer.prefix,
+					''
+				),
+				type: MESSAGE_TYPES.reply.type,
+			};
+		}
```
Comment (kevenleone) [inline modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/ActivityQuestionRow.es.js:72]: I think we should follow the pattern and create another MESSAGE_TYPE for bestAnswer.

My suggestion is to create: 

```
bestAnswer: {type: 4},
```

```suggestion
		if (question.showAsAnswer) {
			return {
				label: Liferay.Language.get('best-answer'),
				symbol: 'check-circle-full',
				text: stripHTML(question.articleBody).replace(
					MESSAGE_TYPES.answer.prefix,
					''
				),
				type: MESSAGE_TYPES.bestAnswer.type,
			};
		}
```

You need to create a language property for best-answer

File: modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/ActivityQuestionRow.es.js [hunk 2]
```diff
@@ -107,17 +123,24 @@ export default function ActivityQuestionRow({
 				/>
 			</Link>
 
-			<div className="c-mb-1 c-mt-2 stretched-link-layer">
+			<div
+				className={classNames('c-mb-1 c-mt-2 stretched-link-layer', {
+					'questions-answer questions-answer-success':
+						messageType.isBestAnswer,
```
Comment (kevenleone) [inline modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/ActivityQuestionRow.es.js:129]:
```suggestion
						messageType.type === MESSAGE_TYPES.bestAnswer.type,
```

File: modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/ActivityUI.es.js [hunk 3]
```diff
@@ -48,7 +48,7 @@ const getQuestionCreatedInDays = (dateCreated) => {
 };
 
 const ActivityHeaderBadge = ({
-	messageType: {label, symbol, type},
+	messageType: {isBestAnswer, label, symbol, type},
```
Comment (kevenleone) [inline modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/ActivityUI.es.js:51]: Remove it

```suggestion
	messageType: {label, symbol, type},
```

File: modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/ActivityUI.es.js [hunk 4]
```diff
@@ -57,7 +57,8 @@ const ActivityHeaderBadge = ({
 		<div className="align-items-center d-flex flex-wrap justify-content-between">
 			<ul className="align-items-center c-mb-2 d-flex flex-nowrap list-badges list-unstyled stretched-link-layer">
 				{DAYS_SINCE_CREATED <= DAYS_UNTIL_SHOW_LABEL &&
-					type !== MESSAGE_TYPES.reply.type && (
+					type !== MESSAGE_TYPES.reply.type &&
+					!isBestAnswer && (
```
Comment (kevenleone) [inline modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/ActivityUI.es.js:61]:
```suggestion
					!type !== MESSAGE_TYPES.bestAnswer.type && (
```

File: modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/ActivityUI.es.js [hunk 5]
```diff
@@ -71,6 +72,7 @@ const ActivityHeaderBadge = ({
 							'bg-light label-secondary text-uppercase',
 							{
 								'questions-reply': symbol === 'reply',
+								'text-success border border-success': isBestAnswer,
```
Comment (kevenleone) [inline modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/ActivityUI.es.js:75]:
```suggestion
								'questions-reply': type === MESSAGE_TYPES.reply.type,
								'text-success border border-success': type === MESSAGE_TYPES.bestAnswer.type,
```

File: modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/ActivityUI.es.js [hunk 6]
```diff
@@ -133,7 +135,10 @@ const ActivityHeader = ({
 	</h5>
 );
 
-const ActivityBody = ({messageType: {symbol, type}, question}) => {
+const ActivityBody = ({
+	messageType: {isBestAnswer, symbol, type},
```
Comment (kevenleone) [inline modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/ActivityUI.es.js:139]:
```suggestion
	messageType: {symbol, type},
```

File: modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/ActivityUI.es.js [hunk 7]
```diff
@@ -153,13 +158,15 @@ const ActivityBody = ({messageType: {symbol, type}, question}) => {
...
 				/>
 
-				<QuestionBadge
-					className="questions-reply"
-					isActivityBadge
-					symbol={symbol}
-					symbolClassName="questions-comment-reply-icon"
-					value={stripHTML(question.articleBody)}
-				/>
+				{!isBestAnswer && (
```
Comment (kevenleone) [inline modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/ActivityUI.es.js:161]:
```suggestion
				type !== MESSAGE_TYPES.bestAnswer.type && (
```

File: modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/ArticleBodyAnwser.es.js [hunk 8]
```diff
@@ -24,6 +25,7 @@ export default function ArticleBodyAnwser({
 	encodingFormat,
 	hasCompanyMx,
 	id,
+	isBestAnswer = false,
```
Comment (kevenleone) [inline modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/ArticleBodyAnwser.es.js:28]: Remove it

```suggestion
	
```

File: modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/ArticleBodyAnwser.es.js [hunk 9]
```diff
@@ -38,7 +40,12 @@ export default function ArticleBodyAnwser({
 	const _companyName = hasCompanyMx && companyName ? `(${companyName})` : '';
 
 	return (
-		<div className="h-100 p-3 position-relative rectangle-comment w-100">
+		<div
+			className={classNames(
+				'h-100 p-3 position-relative rectangle-comment w-100',
+				{'questions-answer questions-answer-success': isBestAnswer}
```
Comment (kevenleone) [inline modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/ArticleBodyAnwser.es.js:46]:
```suggestion
				{'questions-answer questions-answer-success': type === MESSAGE_TYPES.bestAnswer.type}
```

File: modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/utils/client.es.js [hunk 10]
```diff
@@ -821,6 +822,10 @@ export const getUserActivityQuery = `
 						title
 					}
 				}
+				parentMessageBoardMessage {
+					headline
+					articleBody
+				}
```
Comment (kevenleone) [inline modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/utils/client.es.js:828]: I didn't found this parentMessageBoardMessage being used, this is needed?

**Review by kevenleone — CHANGES_REQUESTED** [review]
The code looks good, but I left suggestions for you according to the created pattern

@juliocezar96

**Comment by kevenleone** [comment]
Just starting reviewing :)

**Comment by kevenleone** [comment]
resent: https://github.com/pt-liferay-solutions/liferay-portal/pull/2832

**Pull request closed without merging**